sequential/dpm.py

```diff
@@ -0,0 +1,141 @@
+
+    # Drop frequencies in the end of mined patterns
+    patterns_pos = drop_frequency(patterns_pos)
+    patterns_neg = drop_frequency(patterns_neg)
+
+    aggregation_to_patterns = dict()
+
+    intersection_patterns = set(map(tuple, patterns_pos)).intersection(set(map(tuple, patterns_neg)))
+    aggregation_to_patterns[DichotomicAggregation.intersection] = sorted(list(map(list, intersection_patterns)))
+
+    union_patterns = set(map(tuple, patterns_pos)).union(set(map(tuple, patterns_neg)))
+    aggregation_to_patterns[DichotomicAggregation.union] = sorted(list(map(list, union_patterns)))
+
+    unique_neg_patterns = set(map(tuple, patterns_neg)) - set(map(tuple, patterns_pos))
+    aggregation_to_patterns[DichotomicAggregation.unique_neg] = sorted(list(map(list, unique_neg_patterns)))
+
+    unique_pos_patterns = set(map(tuple, patterns_pos)) - set(map(tuple, patterns_neg))
+    aggregation_to_patterns[DichotomicAggregation.unique_pos] = sorted(list(map(list, unique_pos_patterns)))
+
+    return aggregation_to_patterns
+
+
+def get_one_hot_encodings(sequences: List[list], patterns: List[list], constraints: Union[list, None] = None,
+                          rolling_window_size: int = 10, drop_pattern_frequency=False) -> pd.DataFrame:
+    """
+    Create a data frame having one-hot encoding of sequences.
+
+    Parameters
+    ----------
+    sequences: List[list]
+        A list of sequences of items.
+    patterns: List[list]
+        A list of interested patterns, which defines the encoding space.
+    constraints: Union[list, None]
+        The constraints enforced in the creation of encoding
+    rolling_window_size: int
+        The rolling window along a sequence within which patterns are detected locally. It controls the length of
+        sequence subject to the pattern detection, to speed up the encodings generation.
+        (rolling_window_size=10 by default).
+    drop_pattern_frequency: bool
+        Drop the frequency appended in the end of each input pattern, drop_pattern_frequency=False by default.
+
+    Returns
+    -------
+    A data frame having one-hot encoding of sequences using the given patterns.
+
+    """
+
+    # Drop the frequency appended
+    if drop_pattern_frequency:
+        patterns = drop_frequency(patterns)
+
+    if isinstance(sequences[0][0], str):
+        check_true(not isinstance(patterns[0][-1], int),
+                   ValueError("Patterns should not contain integers! "
+                              "Check if the frequency is appended to the end of given patterns."))
+        check_true(isinstance(patterns[0][0], str),
+                   ValueError("When items are strings, patterns should also be strings."))
+
+        str_to_int, int_to_str = item_map(sequences)
+        sequences = string_to_int(str_to_int, sequences)
+        patterns = string_to_int(str_to_int, patterns)
+
+    df = pd.DataFrame()
+    df['sequence'] = sequences
+    df['seq_ind'] = list(range(len(sequences)))
```

Comment:
minor comment: why need indexes as a new column? Cannot we use DF indexes? OR they are not the same?

Reply:
Right, indexes would suffice. Just tried to use indexes and it worked.